src/App.js

```diff
@@ -14,11 +15,95 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goodsList: [],
+    selectedGood: 'Jam',
+  }
+
+  close() {
+    this.setState({ goodsList: [] });
+  }
+
+  option(event, good) {
+    const { target } = event;
+    const item = target.closest('.goods-list__good');
+
+    if (item.classList.contains('selected')) {
+      const index = this.state.goodsList.findIndex(pos => pos === good);
+
+      this.setState((prevState) => {
+        const updatedList = [...prevState.goodsList];
+
+        updatedList.splice(index, 1);
+
+        return {
+          ...prevState, goodsList: updatedList,
+        };
+      });
+    } else {
+      this.setState(prevState => ({
+        ...prevState, goodsList: [...prevState.goodsList, good],
+      }));
+    }
+  }
+
+  render() {
+    const { goodsList } = this.state;
+    let selectedItems;
+
+    if (goodsList.length) {
+      if (goodsList.length === 1) {
+        selectedItems = `${goodsList.join(', ')} is selected`;
+      } else {
+        selectedItems = `${goodsList.join(', ')} are selected`;
+      }
+    } else {
+      selectedItems = 'No goods selected';
+    }
+
+    return (
+      <div className="App">
+        <h1 className="title">
+          {selectedItems}
+          <button
+            type="button"
+            className="close"
+            onClick={() => {
+              this.close();
+            }}
+          >
+            X
+          </button>
+        </h1>
+        <ul className="goods-list">
+          {goodsFromServer.map(good => (
+            <li
+              key={good}
+              className={
+                classNames(`goods-list__good ${good}`,
+                  { selected: this.state.goodsList.includes(good) })
```

Comment:
you can create some variable (`isIncluded` for example) inside callback in `map` method before you return `<li>`, and use this variable in all places where you need it. with this approach, you can use the method includes only once